utils/keeper-bench/Runner.cpp

```diff
@@ -328,9 +370,519 @@ bool Runner::tryPushRequestInteractively(Coordination::ZooKeeperRequestPtr && re
+            return std::nullopt;
+        return static_cast<Int32>(field.safeGet<Int64>());
+    }
+
+    std::optional<Coordination::Error> getError(size_t row) const
+    {
+        auto field = getField(error_idx, row);
+        if (field.isNull())
+            return std::nullopt;
+
+        return static_cast<Coordination::Error>(field.safeGet<Int64>());
+    }
+private:
+    DB::Field getField(size_t position, size_t row) const
+    {
+        DB::Field field;
+        block.getByPosition(position).column->get(row, field);
+        return field;
+    }
+
+    DB::Block block;
+    size_t hostname_idx = 0;
+    size_t request_event_time_idx = 0;
+    size_t thread_id_idx = 0;
+    size_t session_id_idx = 0;
+    size_t xid_idx = 0;
+    size_t has_watch_idx = 0;
+    size_t op_num_idx = 0;
+    size_t path_idx = 0;
+    size_t data_idx = 0;
+    size_t is_ephemeral_idx = 0;
+    size_t is_sequential_idx = 0;
+    size_t response_event_time_idx = 0;
+    size_t error_idx = 0;
+    size_t requests_size_idx = 0;
+    size_t version_idx = 0;
+};
+
+struct RequestFromLog
+{
+    Coordination::ZooKeeperRequestPtr request;
+    std::optional<Coordination::Error> expected_result;
+    int64_t session_id = 0;
+    size_t executor_id = 0;
+    bool has_watch = false;
+    DB::DateTime64 request_event_time;
+    DB::DateTime64 response_event_time;
+    std::shared_ptr<Coordination::ZooKeeper> connection;
+};
+
+struct ZooKeeperRequestFromLogReader
+{
+    ZooKeeperRequestFromLogReader(const std::string & input_request_log, DB::ContextPtr context)
+    {
+        std::optional<DB::FormatSettings> format_settings;
+
+        file_read_buf = std::make_unique<DB::ReadBufferFromFile>(input_request_log);
+        auto compression_method = DB::chooseCompressionMethod(input_request_log, "");
+        file_read_buf = DB::wrapReadBufferWithCompressionMethod(std::move(file_read_buf), compression_method);
+
+        DB::SingleReadBufferIterator read_buffer_iterator(std::move(file_read_buf));
+        auto [columns_description, format] = DB::detectFormatAndReadSchema(format_settings, read_buffer_iterator, context);
+
+        DB::ColumnsWithTypeAndName columns;
+        columns.reserve(columns_description.size());
+
+        for (const auto & column_description : columns_description)
+            columns.push_back(DB::ColumnWithTypeAndName{column_description.type, column_description.name});
+
+        header_block = std::move(columns);
+
+        file_read_buf
+            = DB::wrapReadBufferWithCompressionMethod(std::make_unique<DB::ReadBufferFromFile>(input_request_log), compression_method);
+
+        input_format = DB::FormatFactory::instance().getInput(
+            format,
+            *file_read_buf,
+            header_block,
+            context,
+            context->getSettingsRef().max_block_size,
+            format_settings,
+            1,
+            std::nullopt,
+            /*is_remote_fs*/ false,
+            DB::CompressionMethod::None,
```

Comment:
If we specify method here we can avoid `wrapReadBufferWithCompressionMethod`

Reply:
is there a benefit for that?
this seems more flexible